webpack/webpack.prod.js

```diff
@@ -1,17 +1,62 @@
-const merge = require('webpack-merge');
-const common = require('./webpack.common');
+const merge = require('webpack-merge')
+const common = require('./webpack.common')
+const MiniCssExtractPlugin = require('mini-css-extract-plugin')
+const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin')
+const HtmlWebpackPlugin = require('html-webpack-plugin')
+const path = require('path')
+const CleanWebpackPlugin = require('clean-webpack-plugin')
 
 module.exports = merge(common, {
   mode: 'production',
 
+  devtool: 'source-map',
+
   output: {
     filename: '[name].[chunkhash].js',
     chunkFilename: '[name].[chunkhash].chunk.js',
-    publicPath: '/',
+    publicPath: '/'
   },
 
```

Comment:
Add optimization with TerserWebpackPlugin along with OptimizeCSSAssetsPlugin:
```
optimization: {
    minimizer: [
      new TerserJSPlugin({}),
      new OptimizeCSSAssetsPlugin({})
    ]
  },
```

Reply:
I saw people use `splitChunk` in `optimization` as well: 
https://webpack.js.org/plugins/split-chunks-plugin/
Haven't gone thru its real benefits but probably 1 consideration

Reply:
Webpack 4 has mode feature, when set it to production, it has TerserJSPlugin() by default. No need to override optimization to include TerserJSPlugin

https://webpack.js.org/configuration/mode/#mode-production